tests/drum/test_units.py

```diff
@@ -975,3 +976,180 @@ def test_binary_class_labels_from_target():
+                ],
+            },
+            1,
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "int", "min": 0, "max": 1},
+                ],
+            },
+            0,
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "float", "min": 0.0, "max": 1.0, "default": 1.0},
+                ],
+            },
+            1.0,
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "float", "min": 0.0, "max": 1.0},
+                ],
+            },
+            0.0,
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "string", "default": "string"},
+                ],
+            },
+            "string",
+        ),
+        ({ModelMetadataKeys.HYPERPARAMETERS: [{"name": "param_name", "type": "string"}]}, "",),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {
+                        "name": "param_name",
+                        "type": "select",
+                        "values": ["value 1", "value 2"],
+                        "default": "value 2",
+                    }
+                ],
+            },
+            "value 2",
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "select", "values": ["value 1", "value 2"],}
+                ],
+            },
+            "value 1",
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {
+                        "name": "param_name",
+                        "type": "multi",
+                        "values": {
+                            "int": {"min": 0, "max": 1},
+                            "select": {"values": ["value 1", "value 2"]},
+                        },
+                        "default": "value 2",
+                    }
+                ],
+            },
+            "value 2",
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {
+                        "name": "param_name",
+                        "type": "multi",
+                        "values": {
+                            "int": {"min": 0, "max": 1},
+                            "select": {"values": ["value 1", "value 2"]},
+                        },
+                    }
+                ],
+            },
+            0,
+        ),
+        (
+            {
+                ModelMetadataKeys.HYPERPARAMETERS: [
+                    {"name": "param_name", "type": "multi", "values": {},}
+                ],
+            },
```

Comment:
I think we should throw an exception if this is present in the params file. Pretty sure we error in DR when this happens as well